src/main/java/com/infoshareacademy/controller/RecipeController.java

```diff
@@ -31,11 +32,29 @@ public RecipeController(RecipeService recipeService) {
     }
 
     @GetMapping
+    public String listRecipesWithPagination(Model model) {
+        return listOfPage(model,1);
+    }
+
+    @GetMapping("/page/{pageNumber}")
+    public String listOfPage(Model model, @PathVariable("pageNumber") int currentPage) {
+        Page<Recipe> page = recipeService.getAllRecipeWithPagination(currentPage);
+        long totalRecipes = page.getTotalElements();
+        int totalPages = page.getTotalPages();
+
+        model.addAttribute("currentPage", currentPage);
+        model.addAttribute("totalRecipes", totalRecipes);
+        model.addAttribute("totalPages", totalPages);
+        model.addAttribute("recipes", page);
+        return "recipes";
+    }
+
     public String getAllRecipeButCanFilterByMealType(Model model, @Param("meal") Meal meal) {
         model.addAttribute("meal", meal);
         model.addAttribute("recipes", recipeService.getRecipesByMeal(meal));
         return "recipes";
     }
+    
```

Comment:
To trzeba połączyć, plus:
        Page<Recipe> page = recipeService.getAllRecipeWithPagination(currentPage);
        long totalRecipes = page.getTotalElements();
        int totalPages = page.getTotalPages();
to do modelu można dać żeby skrócić.